ios/engine/KMEI/KeymanEngine/Classes/Keyboard/InputViewController.swift

```diff
@@ -333,6 +333,60 @@ open class InputViewController: UIInputViewController, KeymanWebDelegate {
       }
     }
   }
+  
+  func deleteSelection() -> Bool {
+    if let selected = textDocumentProxy.selectedText, selected.count > 0 {
+      /*
+        Since we're doing some funky text manipulation, it's best to add
+        a "canary" check in case something does go awry with it in
+        the future.
+      */
+      let beforeManipulation = textDocumentProxy.documentContextBeforeInput ?? ""
+
+      /*
+        Apple has some special nuances to its text deletion that our internal
+        Web engine does not emulate.
+       
+        .deleteBackward() behaviors:
+        - If there is selected text immediately following a space (U+0020),
+          it will delete that space IN ADDITION to the selected text.
+        - If there is selected text that starts mid-word, it will NOT delete
+          the preceding character.
```

Comment:
This seems to be just one special case for space preceding a selection, not two special cases. The way this is written it looks like there are two special cases.

Remember that our policy on backspace with selected text is to delete the selection, but not the preceding character (#9073).

Reply:
```suggestion
        We have a problem to resolve here: we cannot simply delete the selection 
        with either .deleteBackward() or .insertText(""):
        
        - If there is selected text immediately following a space (U+0020),
          .deleteBackward() will delete that space IN ADDITION to the selected text.
        - Unlike .insertText("-any-string-here"), .insertText("") does nothing -- 
          it does not replace the selection with the new string.
```